### PR TITLE
Add Trino 388 release notes

### LINKED ISSUE #12945: Release notes for 388

### PR DESCRIPTION
**Pull request by colebow**
## Description

Assemble the release notes for the upcoming Trino 388 release.

> Is this change a fix, improvement, new feature, refactoring, or other?

Improvement

> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)

Docs only.

> How would you describe this change to a non-technical end user or system administrator?

Add the release notes for Trino 388 with sections for general, security, connectors and other topics as included in the release.

## Related issues, pull requests, and links

Fixes #12945

## Documentation

( ) No documentation is needed.
(✅) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

(✅) No release notes entries required.
( ) Release notes entries required with the following suggested text:

## Verification for each commit

Format: commit, committer/maintainer, PR/issue number, ✅ / ❌ rn ✅ / ❌ docs
✅ rn - release note added and verified, or assessed to be not necessary, set to ❌ rn before completion
✅ docs - need for docs assessed and merged, or assessed to be not necessary, set to ❌ docs before completion

## 22 June 2022 ✅ 

a3ea92b4fc4e9858cc2969952137c4257abfcc1d @ebyhr #12946 ✅ rn ✅ docs

## 23 June 2022 ✅ 

3d65abe7b71e943e3183c773bcfaf14b9d6a8ca6 @losipiuk #12930 ✅ rn ✅ docs
4fc6f441340014b9ea50752288086a054e2821e7 @ebyhr #12955 ✅ rn ✅ docs
2c5492176fcf5cfbf7343ad96b8b1d8f79d2c488 @raunaqmorarka #12926 ✅ rn ✅ docs
9bd9dc3d57d90f9f6cb36d2d87f49181e7cf67b0 @raunaqmorarka #12926 ✅ rn ✅ docs
a815a6f08d520fd8f059f29ef98571110bcddb15 @arhimondr #12443 ✅ rn ✅ docs
f331e59c36345de2112c56756557d8ade8fa829e @arhimondr #12443 ✅ rn ✅ docs
f0cdc0270e653ba4f7cbc278004a0281ed20ab4f @arhimondr #12443 ✅ rn ✅ docs
7d0c3447ec4c77238bd2384ebc91f3b9865a34ca @arhimondr #12443 ✅ rn ✅ docs
726fbd77f59ebff1885f58819d3dbdbd23f41708 @arhimondr #12443 ✅ rn ✅ docs
5eb47b1e1bb1b804af142c6719fe80d6f0c926b0 @findepi #12900 ✅ rn ✅ docs
0cc695970056fb21eeeef6ca33f43e1ffac9cb5f @findepi #12900 ✅ rn ✅ docs
67656139e8a292bb9251e8706bf497600595291d @arhimondr #12753 ✅ rn ✅ docs
e4654afaabb2202d13f7c2eb60513a850770df9f @findepi #12960 ✅ rn ✅ docs
efb928edfb880b4553edb164c98927a1cdbaac92 @ebyhr #12932 ✅ rn ✅ docs

## 24 June 2022 ✅ 

c484ba21f7162466b94c9359e7f1553cc1ef5967 @raunaqmorarka #12694 ✅ rn ✅ docs
b814e35904eb1c1f7dad3c201f47f0f8b44b76e0 @raunaqmorarka #12694 ✅ rn ✅ docs
45decebe4f0580c82d2be69210b8e45783620fb6 @raunaqmorarka #12694 ✅ rn ✅ docs
8d0946c81b0995bc96023b56ec7149a44462efdd @raunaqmorarka #12694 ✅ rn ✅ docs
593a4f0c9b3c66e5fb7442362a62804f835e62e1 @raunaqmorarka #12694 ✅ rn ✅ docs
2ed5041d05a33144756d1837e07029ec65a1b781 @raunaqmorarka #12694 ✅ rn ✅ docs
9e206fda398f443dd884a7b49ff184a0eaf3e493 @raunaqmorarka #12694 ✅ rn ✅ docs
ae978622ed8efebd3274b431ea891628eea8c2bf @raunaqmorarka #12694 ✅ rn ✅ docs
faad0de33effa911e36f3310685e9f9e8e8a9183 @findepi #12919 ✅ rn ✅ docs
8e84563e9158033392328377f3a2ed60994ee52e @findepi #12919 ✅ rn ✅ docs
6f4740e472f11f988540f5a6a7b8b8d3760dbe6e @findepi #12961 ✅ rn ✅ docs
8cb79f08493c949f75b48fc102475ede15e331de @findepi #12959 ✅ rn ✅ docs
bfe5619ebfa5d672bc7da70198def3bf9fc66dcb @ebyhr #12675 ✅ rn ✅ docs
07713949c5b37e120d52ed3cbad85e165d777368 @findepi #12978 ✅ rn ✅ docs
b0c0dfaa2b9c85c647f88e77b47c30eff2c9b6f0 @raunaqmorarka #12857 ✅ rn ✅ docs
f4c6a189e20902874f68fa49c5eb56ee36c5423c @raunaqmorarka #12857 ✅ rn ✅ docs
ddd7cb1e9b266cb11c687985e9fb070f07b21866 @raunaqmorarka #12857 ✅ rn ✅ docs
4e37e95cf313a0b249d20afebbd020f3bc3cb115 @raunaqmorarka #12857 ✅ rn ✅ docs
ff166e3e6e5d2b52545923390cd9c6d18fbe25f3 @raunaqmorarka #12857 ✅ rn ✅ docs
766c0ac41fb6ba11ff11d71a57e10939b211c0ed @raunaqmorarka #12857 ✅ rn ✅ docs
8f5d68963c33ef6eb75068fa33b5c096c4088169 @raunaqmorarka #12857 ✅ rn ✅ docs
33db28e411fcaff250d301bd32d3490a4d8f8636 @raunaqmorarka #12857 ✅ rn ✅ docs
83734c37636bc5ab9839535791773244ac0ee0da @martint #12982 ✅ rn ✅ docs
24a59cee69dc89eeb3b170bbdf76f5001742b882 @ebyhr #12089 ✅ rn ✅ docs

## 25 June 2022 ✅ 

9c382d2248f47ec87af190062146bdfe0e678dca @raunaqmorarka #12591 ✅ rn ✅ docs
937fe89d2ec298fa60ae7b79c31abce51f573656 @hashhar #11263 ✅ rn ✅ docs
4f629df12b0e41cf6dcb6e33e4899b264de63590 @findepi #11401 ✅ rn ✅ docs
0ee093027c3e05eac74fa0a0fe173e3b5cd126db @findepi #11401 ✅ rn ✅ docs
c1944d92189d64f9c7cb4ad679f03d7bde233470 @dain #12803 ✅ rn ✅ docs
d7cb84877fa2fa34e9a17985012bf35b50cded5a @dain #12803 ✅ rn ✅ docs
5febea2f0a0ed5b3206336ae999a9c1da52333bc @dain #12803 ✅ rn ✅ docs
72a934d4a054254d8fe9cab91b167d2012893646 @dain #12803 ✅ rn ✅ docs
e5ec024b2b3a663b18a1ee49a14c707419e7e148 @dain #12803 ✅ rn ✅ docs
9e517988e1ef4e88c97c259f03972ff8b76358cf @dain #12803 ✅ rn ✅ docs
c71db2f7b4f1a16139ae147098cffdfc4b66c517 @dain #12803 ✅ rn ✅ docs
feb3309ac137d57916c7755b931b44c2d65f113e @dain #12803 ✅ rn ✅ docs
cdb9f6218843075eda5537c18a6c02f4d6285e2d @dain #12803 ✅ rn ✅ docs
ac1fd52b1b237f08ad940dcfdc4a3e38593222d0 @dain #12803 ✅ rn ✅ docs
2721ad20dd4aad89f6a8e16bd8baae3ddb0c57d7 @dain #12803 ✅ rn ✅ docs
64c0ee633cdeb58f14a0240661865cbd4f6135ef @dain #12803 ✅ rn ✅ docs
eb82de23888aa042cce06699bbafd0a2f9e1943f @dain #12803 ✅ rn ✅ docs
29754b80821c1985c0d3d37614f91de42fc1e551 @dain #12803 ✅ rn ✅ docs
9a9f39a562c4075ec7e85e45315e74d8de4eb52b @dain #12806 ✅ rn ✅ docs
794d55fc204090e0f0a594e31e321679c65a096e @dain #12806 ✅ rn ✅ docs
814539fee798f159233103e23eceb95df21682f6 @dain #12806 ✅ rn ✅ docs
999331d11fe1dfc53f5f575c2b3441cbf81489eb @dain #12806 ✅ rn ✅ docs
c9f881f955147d6b1a4dee93e9bb64e41e301cd3 @dain #12806 ✅ rn ✅ docs

## 27 June 2022 ✅ 

f5a887db1ea3dcf906a1636790b7b4027cb2cdf8 @raunaqmorarka #12988 ✅ rn ✅ docs
5a6fb213451e8d5c6fe2b4b56ed0c2c54e3769d1 @losipiuk #12967 ✅ rn ✅ docs
b91476f5fdf475893d797671d00d48c166ed1f49 @Praveen2112 #12664 ✅ rn ✅ docs
1a1b831f27dac4aee05ad59e5aeccbb7f47987fd @Praveen2112 #12664 ✅ rn ✅ docs
b3d03cde33583e312f077d53b9133e13487ce92b @Praveen2112 #12664 ✅ rn ✅ docs
232d139abd51adcd7c618e1f6f1fe4c2197cf2d7 @findepi #12979 ✅ rn ✅ docs
d746be11a3d467d6c0ce1c8f366bd408a43a0f8c @findepi #12997 ✅ rn ✅ docs
98523020c8cb92790f9dadb66e8422da40fa40b1 @kasiafi #12999 ✅ rn ✅ docs
faea3a841efbd421a1757ab6bd48275db6e35fae @kasiafi #12999 ✅ rn ✅ docs
a405994a4029fc2e654950abcba602320897a60c @raunaqmorarka #12934 ✅ rn ✅ docs
4c61680aee82e8626a9fe7face9127ede58e8393 @raunaqmorarka #12934 ✅ rn ✅ docs

## 28 June 2022 ✅ 

6b95dc70766636a2367ba249f37e7df3946aa452 @sopel39 #12888 ✅ rn ✅ docs
ef39062dfcb0bdaac7dd96583530bc312e1f316f @findepi #13010 ✅ rn ✅ docs
56677a7302586d617cc4ff4d7856f68e8342f3a2 @findepi #13010 ✅ rn ✅ docs
caa6ed6578735cb747e9d8c098ab5849894e0387 @findepi #13011 ✅ rn ✅ docs
a09436036ae3d10ea2dcb98ae973e8933f3e0646 @findepi #13011 ✅ rn ✅ docs
a50fc40bce55c1b616320a17aa5efd59c804646d @losipiuk #12916 ✅ rn ✅ docs
c761977c45eeea3eb11a0b32e11dc7eac7df3b9a @losipiuk #12916 ✅ rn ✅ docs
a659c3e41d5560ec7b561b465b81d406eab937de @losipiuk #12916 ✅ rn ✅ docs
554b4b45faa503afb6318642c48adb05150bc143 @losipiuk #12916 ✅ rn ✅ docs
4c89f548868bae8282333633c5232b11a8db6aff @losipiuk #12916 ✅ rn ✅ docs
b54b7fc49da25950ab3a7dba51804c2d4f9afa4c @losipiuk #12916 ✅ rn ✅ docs
36d625a3c4a7014b9ae0f5a80503a4f68b9abdd0 @losipiuk #12916 ✅ rn ✅ docs
ebaf12655aaaf27d129f228e2ce187ddd208350e @losipiuk #12916 ✅ rn ✅ docs
ced7c97573c95abfbbadd72d281e593c0506e67a @losipiuk #12916 ✅ rn ✅ docs
9bb523186b46be0a429bedfac5e52fc4659b0920 @losipiuk #12916 ✅ rn ✅ docs
b89a4322896c0a788390e7113553b674fd64f453 @findepi #12907 ✅ rn ✅ docs
c2c83194d397cff4daffd28406e45405debeb878 @findepi #12907 ✅ rn ✅ docs
26f4aceb38d45e5a57186e6ad35e990a4a5acd6e @findepi #12907 ✅ rn ✅ docs
69d377df86437e02bf4649c009d732f89d89b0fd @findepi #12907 ✅ rn ✅ docs
0d1954fb288891e19dd51a4e1b54f2f5350106bd @findepi #12907 ✅ rn ✅ docs
d809130c9feec9c03b20629d114a015c483d03db @findepi #12907 ✅ rn ✅ docs
ac8d2d472201eef7c9a3129bd72835f57f2c3faa @kokosing #12998 ✅ rn ✅ docs
dcbf7936a3c283643d7db16ae991f1fc59d35f7a @findepi #13019 ✅ rn ✅ docs
5b41c245fd258a5ff0735d7ed34984bb3128c42c @findepi #12704 ✅ rn ✅ docs
6bba61b07306955cd38a8d373dede39ddfc7873b @findepi #12704 ✅ rn ✅ docs
aacbe2a45db1d525d625160fb8b427a4ea17785b @findepi #12971 ✅ rn ✅ docs
4f04191a712c184ff4a1a49791a281ef13a6aa8c @findepi #12971 ✅ rn ✅ docs
5638b5c994527d9f0ffa8b99a556f0a9fdc0c47d @martint #13022 ✅ rn ✅ docs

## 29 June 2022 ✅ 

ad783ef840f0c6af6fcc1a2333950a8cb44a6650 @ebyhr #13023 ✅ rn ✅ docs
e8a1877b277075e14f1a36c4c13df2c79b421302 @kasiafi #13008 ✅ rn ✅ docs
7f6087b23545029a34c0e0c314109bad7443d317 @findepi #12972 ✅ rn ✅ docs
1aa7c986d48420e86d6e303009cb68470a211f04 @raunaqmorarka #12996 ✅ rn ✅ docs
2b5eea9ae25ae12ad841654bb7d67e3258938d3f @sopel39 #13029 ✅ rn ✅ docs
24438dd7c2f261d4e3af66f3d16e2dee9aa0ed00 @findepi #12879 ✅ rn ✅ docs
c1fa9d4c1339a3a71a46e9bb264c9bc317469bd3 @sopel39 #12927 ✅ rn ✅ docs
0841c0b213a4d085a936cfc06aadbcbce026cfb2 @martint #13006 ✅ rn ✅ docs
ea6cc858c0eb6f878925b2696ddff6723fe08968 @martint #12725 ✅ rn ✅ docs
d4512db88e0761d960976cd0d0a6363555605b1f @martint #13018 ✅ rn ✅ docs
5bc8d88b3d13a620e7c346609494dbcdb7e83800 @martint #13018 ✅ rn ✅ docs
116476562738f567cfb633fb16cb2fc6ecf5837d @martint #13003 ✅ rn ✅ docs

### CODE REVIEW

File: docs/src/main/sphinx/release/release-388.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+## Delta Lake connector
+
+* Add support for setting table and column comments with the `COMMENT`
+  statement. ({issue}`12971`)
+* Support reading tables with the property `delta.columnMapping.mode=name`. ({issue}`12675`)
+* Allow renaming tables with an explicitly set location. ({issue}`11400`)
+
+## Elasticsearch connector
+
+* Remove support for Elasticsearch servers older than 6.6.0. ({issue}`11263`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-388.md:27]:
```suggestion
* Remove support for Elasticsearch versions below 6.6.0. ({issue}`11263`)
```

File: docs/src/main/sphinx/release/release-388.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+## Hive connector
+
+* Improve performance of listing files and generating splits when recursive
+  directory listings are enabled and tables are stored in S3. ({issue}`12443`)
+* Fix incompatibility with Apache Hive when using files produced by optimized
+  parquet writer with timestamp columns. ({issue}`12857 `)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-388.md:34]: by the optimized Parquet writer

File: docs/src/main/sphinx/release/release-388.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+* Prevent reading from a table that has been modified within the same Trino
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-388.md:35]: that was modified

File: docs/src/main/sphinx/release/release-388.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  transaction. Previously, this returned incorrect query results. ({issue}`11769`)
+
+## Iceberg connector
+
+* Add support for reading ORC `tinyint` type. ({issue}`8919`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-388.md:40]:
```suggestion
* Add support for reading `tinyint`-type columns from ORC files. ({issue}`8919`)
```

**Review by mosabua — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/release/release-388.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
+# Release 388 (29 June 2022)
+
+## General
+
+* Add support for `EXPLAIN (TYPE LOGICAL, FORMAT JSON)`. ({issue}`12694`)
+* Add `use_exact_partitioning` session property to re-partition data when the
+  upstream stage's partitioning does not exactly match what the downstream stage
+  expects. ({issue}`12495`)
+* Improve read performance for `ROW` data types. ({issue}`12926`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-388.md:9]: Lower-case `row`

File: docs/src/main/sphinx/release/release-388.md [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+## Security
+
+* Add [refresh token](https://oauth.net/2/refresh-tokens/) support in OAuth 2.0. ({issue}`12664`)
+
+## Delta Lake connector
+
+* Add support for setting table and column comments with the `COMMENT`
+  statement. ({issue}`12971`)
+* Support reading tables with the property `delta.columnMapping.mode=name`. ({issue}`12675`)
+* Allow renaming tables with an explicitly set location. ({issue}`11400`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-388.md:25]: Wouldn't saying "external table" be clearer? cc @findepi

File: docs/src/main/sphinx/release/release-388.md [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+* Improve performance of listing files and generating splits when recursive
+  directory listings are enabled and tables are stored in S3. ({issue}`12443`)
+* Fix incompatibility with Apache Hive when using files produced by the
+  optimized Parquet writer with timestamp columns. ({issue}`12857 `)
+* Prevent reading from a table that was modified within the same Trino 
+  transaction. Previously, this returned incorrect query results. ({issue}`11769`)
+
+## Iceberg connector
+
+* Add support for reading `tinyint`-type columns from ORC files. ({issue}`8919`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-388.md:42]: > ... reading `tinyint` columns ...

File: docs/src/main/sphinx/release/release-388.md [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+* Add the ability to configure the schema for materialized view storage tables. ({issue}`12591`)
+* Remove delete files when running `optimize`. ({issue}`12617`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-388.md:44]: Redundant "delete" in sentence? Or does this refer to the files that track deletions in Iceberg?

Comment (colebow) [reply docs/src/main/sphinx/release/release-388.md:44]: The latter, let me reword. The PR was using "delete files" as a term and I wasn't sure if I could mess with it.

File: docs/src/main/sphinx/release/release-388.md (hunk 6 above)
Comment (colebow) [reply docs/src/main/sphinx/release/release-388.md:25]: I'm honestly not sure what the explicit location means - the PR just says it's adding support for `ALTER TABLE RENAME TO` and doesn't mention any caveats.

**Review by mosabua — APPROVED** [review]
Looks good now.

**Pull request merged**